shotover/src/server.rs

```diff
@@ -277,6 +285,39 @@ impl<C: CodecBuilder + 'static> TcpCodecListener<C> {
             backoff *= 2;
         }
     }
+
+    /// Handle hot reload request by extracting file descriptor and responding
+    async fn handle_hot_reload_request(&mut self, request: HotReloadListenerRequest) {
+        let response = if let Some(ref listener) = self.listener {
+            // Extract the file descriptor from the TcpListener
+            let fd = listener.as_raw_fd();
+
+            // Always parse port from listen_addr to support hostnames and [IPv6]:port formats.
+            let port = self
+                .listen_addr
+                .rsplit_once(':')
+                .and_then(|(_, p)| p.parse::<u16>().ok())
+                .unwrap_or(0);
```

Comment:
`unwrap_or(0)` is concerning.
If its truly impossible to hit this case we should do `.unwrap()`.
If it is possible then we need some kind of error handling rather than returning an invalid port.

Shotover will fail to start if the address does not contain a valid port:
<img width="1146" height="226" alt="image" src="https://github.com/user-attachments/assets/7079e95a-17aa-4ac7-8225-741089fc005c" />
So lets just `.unwrap()` here.